rollup-node.md

```diff
@@ -0,0 +1,285 @@
+from the block's log entries. Refer to the [**L2 deposit feed contract
+specification**][deposit-feed] for details on how deposits are encoded as
+log entries.
+
+[deposit-feed-spec]: TODO
+
+> **TODO LINK** deposit feed contract specification
+
+From these L2 derivation inputs, the rollup node constructs an expanded version
+of the [Engine API PayloadAttributesV1 object][PayloadAttributesV1]:
+
+[PayloadAttributesV1]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
+
+```
+PayloadAttributesOPV1: {
+	timestamp: QUANTITY
+	random: DATA (32 bytes)
+	suggestedFeeRecipient: DATA (20 bytes)
+	transactions: array of DATA
+}
+```
+
+The type notation used here refers to the [HEX value encoding] used by the
+[Ethereum JSON-RPC API specification][JSON-RPC-API], as this structure will need to
+be sent over JSON-RPC. `array` refers to a JSON array.
+
+[HEX value encoding]: https://eth.wiki/json-rpc/API#hex-value-encoding
+[JSON-RPC-API]: https://eth.wiki/json-rpc/API
+
+The object properties must be set as follows:
+
+- `timestamp` is set to the current [unix time] (number of elapsed seconds since
+  00:00:00 UTC on 1 January 1970), rounded to the closest multiple of 2 seconds.
+- `random` is currently set to all zero bytes. After [the merge][merge], the randomness
+  value exposed for the L1 block will be copied in this property.
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, RLP-encoded in the [EIP-2718]
+  format (i.e. as a sequence of a single byte and a byte array), of the allowed
+  types. We specify how to select the transactions below.
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+> For details on RLP encoding, refer to the yellow paper, or look at [this
+> hyperlinked implementation][encode-tx].
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
+
+In particular, we allow the following two transaction types:
+- [EIP-1559]: the current "regular" Ethereum L1 transaction type, which
+  separates gas fees between a base fee and a tip
+- the *[L1 attributes transaction]*, see below
+
+[EIP-1559]: https://eips.ethereum.org/EIPS/eip-1559
+
+Historical Optimistic Ethereum transactions also include [EIP-155] transactions,
+which are not accepted by the network anymore. This is not an [EIP-2718] typed
+transaction, but can be distinguished from those by the fact that their
+RLP-encoding's first byte will always be `>= 0xC0`.
+
+[EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+
+#### Payload Transaction Format
+
+The `transactions` array is filled with the deposits, prefixed by the (single)
+[L1 attributes transaction]. The deposits (which must be similarly RLP-encoded)
```

Comment:
The typed-transaction EIP does not require us to RLP-encode the deposits if we don't want to. And it requires us to prefix it with a type byte. Let's keep the rollup-node doc general, and discuss the deposit format in a separate PR (@maurelian can lead this)